libs/blocks/ost/ost.js

```diff
@@ -75,30 +77,44 @@ export function createLinkMarkup(
   return link;
 }
 
-export default async function init() {
+export default async function init(el) {
+  el.innerHTML = '<div />';
+  const rootElement = el.firstElementChild;
   const aosAccessToken = await getImsToken();
   const country = 'US';
   const language = 'en';
   const environment = 'PROD';
   const wcsApiKey = 'wcms-commerce-ims-ro-user-cc';
   const aosApiKey = 'wcms-commerce-ims-user-prod';
   const checkoutClientId = 'creative';
-  const rootContainer = document.querySelector('.ost');
   const searchParameters = new URLSearchParams(window.location.search);
-  rootContainer.removeChild(rootContainer.firstElementChild);
+
   if (!window.ost) {
     loadStyle(OST_STYLE_URL);
     await loadScript(OST_SCRIPT_URL);
+    await loadStyle('https://use.typekit.net/pps7abe.css');
   }
-  window.ost.openOfferSelectorTool({
-    country,
-    language,
-    environment,
-    wcsApiKey,
-    aosApiKey,
-    aosAccessToken,
-    checkoutClientId,
-    searchParameters,
-    createLinkMarkup,
-  });
+
+  const options = {
+    rootMargin: '0px',
+    threshold: 1.0,
+  };
+
+  const main = document.querySelector('main');
+  const observer = new IntersectionObserver(() => {
+    observer.unobserve(main);
```

Comment:
Wait until Milo displays the block so that we can accurately calculate the height of the container.